jcl/src/java.base/share/classes/com/ibm/oti/vm/J9INTERNAL_CLASS.java

```diff
@@ -0,0 +1,35 @@
+/*[INCLUDE-IF Sidecar17]*/
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package com.ibm.oti.vm;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE})
+public @interface J9INTERNAL_CLASS {
```

Comment:
Odd naming convention for an annotation. Shouldn't we follow class/annotation naming conventions here (`J9InternalClass`)?

Reply:
HotSpot has a similar `@HotSpotIntrinsicCandidate` as an example.

Reply:
It's name similarly to our other internal annoatation:

https://github.com/eclipse/openj9/blob/278be6c0ff617069c6fa2038ffac7afbf4cb5989/jcl/src/java.base/share/classes/java/lang/invoke/VMCONSTANTPOOL_CLASS.java

I'm certainly not married to the name, and we should be moving these things out of com.ibm at some point.

Reply:
Also, I want something fairly obtuse because I'm cheating the annotation lookup in the ClassFileLoadHook code.

Reply:
@DanHeidinga can comment on this during his review.

Reply:
The VMCONSTANTPOOL naming convention is my mess.  I'm not super keen to propagate it.

I'd propose a name that more closely matches the intention and name of the internal bit - something like `J9UnmodifiableClass`?

Reply:
Renamed.